### ISSUE TITLE
Expose stream/List of Events for incompleteEventPublications

### ISSUE THREAD

**Issue by zikozee** [spring-projects/spring-modulith#581]
I am looking at a scenario where I can dynamically batch the incomplete events which can also give room to do some extra work before or after.

**Comment by patrzalektom** [comment]
Mentioned pull request: https://github.com/spring-projects/spring-modulith/pull/616

**Comment by kalwinskidawid** [comment]
@odrotbohm Will it be implemented in the near future? Additionally, once it's possible to use findAll and obtain the UUID, it would be great to also have the ability to call `resubmitIncompletePublication(UUID)`

**Comment by odrotbohm** [comment]
@kalwinskidawid — Can you elaborate on the use case you have in mind?

**Comment by kalwinskidawid** [comment]
@odrotbohm Ideally, no event should go unprocessed, but we don't live in a perfect world. Therefore, in my case, I think it would be useful to have a list of events that didn’t complete successfully available at the service level. That way, we could directly attempt to process them again using their UUIDs.

**Comment by odrotbohm** [comment]
What I wonder is: what would you want to do with the publication beyond resubmitting it? If there's nothing else, isn't the general resubmission mechanism already present, sufficient?

**Comment by kalwinskidawid** [comment]
Yeah, the mechanism already exists, but I think it would be good to know what didn't go through before calling `incompleteEventPublications.resubmitIncompletePublicationsOlderThan(olderThan);`

**Comment by patrzalektom** [comment]
What I had in mind when grabbing that issue and opening PR (https://github.com/spring-projects/spring-modulith/pull/616) is that can be useful for some kind of reporting, or deciding if we want to resubmit specific publications or not. I did not add any tests for that though.

@odrotbohm let me know if that topic would be implemented - I'm more than eager to work further on this 😸

**Comment by raviprakashmishra** [comment]
> What I wonder is: what would you want to do with the publication beyond resubmitting it? If there's nothing else, isn't the general resubmission mechanism already present, sufficient?

One example usecase, I don't want to resubmit all the incomplete all the times because of few reasons -
* Some events might be consistent failures, I may want to skip them if they are too old. I may want to debug and process them manually.
* It's possible that too many events are in failed stage because of external system outage and I may want to process only 100 of them i.e batch size, not too overwhelm the external system.
